src/main/java/com/cloudbees/hudson/plugins/folder/computed/ChildObserver.java

```diff
@@ -58,4 +63,20 @@
      */
     public abstract void created(I child);
 
+    /**
+     * Returns a copy of the item names that have been observed.
+     *
+     * @return a copy of the item names that have been observed.
+     * @since FIXME
+     */
+    public abstract Set<String> observed();
+
+    /**
+     * Returns a copy of the map of orphaned items keyed by name.
+     *
+     * @return a copy of the map of orphaned items keyed by name.
+     * @since FIXME
+     */
+    public abstract Map<String,I> orphaned();
```

Comment:
This is to do with eventing... I cannot give the example of views without the eventing support and the eventing support requires this in the API.

While strictly true that it breaks backwards compatibility, the `ChildObserver` interface is only implemented within `ComputedFolder` and both of those implementations have the corresponding implementations. The other option would have been to change the contract of `ChildObserver` so that the `created` method is not the entry-point and instead use a different entry point for creating (or make `created` final and have that track the observed entries) either of which would be a bigger breaking change.

For any other people that happen to have used the `ChildObserver` abstract class, prior they would not have had the method available, so they would not have been able to call it... 

While we do call this method from `updateChildren`, that case is always using `FullReindexChildObserver` which has the method implemented.